src/libaktualizr/uptane/tuf_test.cc

```diff
@@ -85,6 +88,181 @@ TEST(Role, InvalidDelegationName) {
   EXPECT_THROW(Uptane::Role::Delegation("timestamp"), Uptane::Exception);
 }
 
+Json::Value generateTarget(const std::string& hash, const int length) {
+  Json::Value target;
+  Json::Value hashes;
+  hashes["sha256"] = hash;
+  target["hashes"] = hashes;
+  target["length"] = length;
+  return target;
+}
+
+Json::Value generateDirectorTarget(const std::string& hash, const int length,
+                                   const std::map<Uptane::EcuSerial, Uptane::HardwareIdentifier>& ecu_map) {
+  Json::Value target = generateTarget(hash, length);
+  Json::Value custom;
+  Json::Value ecus;
+  for (auto it = ecu_map.cbegin(); it != ecu_map.cend(); ++it) {
+    Json::Value current;
+    current["hardwareId"] = it->second.ToString();
+    ecus[it->first.ToString()] = current;
+  }
+  custom["ecuIdentifiers"] = ecus;
+  target["custom"] = custom;
+  return target;
+}
+
+Json::Value generateImagesTarget(const std::string& hash, const int length,
+                                 const std::vector<Uptane::HardwareIdentifier>& hardwareIds) {
+  Json::Value target = generateTarget(hash, length);
+  Json::Value custom;
+  Json::Value hwids;
+  for (Json::Value::ArrayIndex i = 0; i < hardwareIds.size(); ++i) {
+    hwids[i] = hardwareIds[i].ToString();
+  }
+  custom["hardwareIds"] = hwids;
+  target["custom"] = custom;
+  return target;
+}
+
+/* Equivalent metadata generated by both repos should match. */
+TEST(Target, Match) {
+  Uptane::HardwareIdentifier hwid("fake-test");
+  std::vector<Uptane::HardwareIdentifier> hardwareIds;
+  std::map<Uptane::EcuSerial, Uptane::HardwareIdentifier> ecu_map;
+  hardwareIds.emplace_back(hwid);
+  ecu_map.insert({Uptane::EcuSerial("serial"), hwid});
+  Uptane::Target target1("abc", generateDirectorTarget("hash_good", 739, ecu_map));
+  Uptane::Target target2("abc", generateImagesTarget("hash_good", 739, hardwareIds));
+  EXPECT_TRUE(target1 == target2);
```

Comment:
Should we also test `target2 == target1`?

Reply:
Probably not a bad idea. They should be identical, but it would help prevent potential mistakes to future edits to the Target class.

Reply:
Yes I would be more comfortable if all the tests checked the reflexivity.